### ISSUE TITLE
Left-side Frame Reloads every click

### ISSUE THREAD

**Issue by jimfoltz** [SketchUp/ruby-api-docs#16]
Does the left side menu reload every click? It appears to, and the docs are kinda of slow as a result even when viewing from local disc.

**Comment by thomthom** [comment]
It's based on the default YARD template. We didn't write one from scratch. This would probably be a much bigger task - most likely a complete refactor.

I'll see if I can get our template customisation uploaded to GitHub. That way people can play around with it.

**Comment by thomthom** [comment]
I just made a dump of our template: https://github.com/SketchUp/sketchup-yard-template (In case anyone wants to play around.)

Note that I didn't have time to test it. Also since we generate from our CPP files, this would have to be run against something else. We're also working on generating stubs - which should be usable. More to come.

**Comment by jimfoltz** [comment]
I appreciate it. However it will be better if I just scrape the html files in my local clone of this repo. Then I can rebuild the links anyway I want and be able to like directly to the classes and methods.

I have been using a Firefox sidebar bookmark to create my own index for some time anyway. 

![2016-11-24_082605](https://cloud.githubusercontent.com/assets/137514/20600069/b434c696-b21f-11e6-9549-9ba14df97085.png)

**Comment by thomthom** [comment]
Ah, gotcha. That's where the other request for anchor links comes in, right?

**Comment by jimfoltz** [comment]
> Ah, gotcha. That's where the other request for anchor links comes in, right?

Correct.  And then I should be able to patch some files to disable the left `iframe` nav altogether and use my sidebar index.

**Comment by thomthom** [comment]
At some point I think YARD used to have a frames option, but now it looks like it just redirect to the single page solution. Not sure what the history is there.

I made good progress of creating Ruby stubs from our CPP source. That means you should be able to run YARD on the stubs which gives us a lot of useful tools to process the API. It becomes easy to generate custom output based on the YARD database.

**Comment by thomthom** [comment]
@jimfoltz  FYI: I just pushed an (incomplete) set of API stubs to a new repo: https://github.com/SketchUp/ruby-api-stubs

Still working on this, but thought I'd just show what we manage to create so far. Figured I'd ping you since you've been maintaining that other repo manually.

**Comment by jimfoltz** [comment]
I'm a little disappointed in Firefox - Chrome runs the site noticeably quicker.

**Comment by MSP-Greg** [comment]
@thomthom, @jimfoltz 

Been a while.  Added c parsing updates to YardT2, and it's always used XMLHttpRequest, so it won't allow static local loading of files (always needs a server, easy to spin one up with Ruby).

See [Ruby Core](https://msp-greg.github.io/ruby_stable/Core/index.html).  'Home' shows other libraries and some 'help' type docs.

I have only tested with 1920 monitors and non-retina iPad.  I'm sure there are issues with other resolutions & phones (why look at a code library on a phone? 80 characters?)...

**Comment by thomthom** [comment]
@jimfoltz the slowness, is this with SU's docs only? Or do you also see it in plain YARD docs? Like SKUI: http://www.rubydoc.info/github/thomthom/SKUI/

We for the most part use YARD's default template, with some minor header and color customisation.
